### ISSUE TITLE
Non-ipv6 addresses are encoded within square brackets

### ISSUE THREAD

**Issue by simonnineone** [softwaremill/sttp#195]
Hello,

I am experiencing a problem that I believe is a bug and sttp treats non-ipv6 addresses as if they were IPv6.

Addresses like abc get encoded as [abc]:
```scala
scala> uri"abc"
res0: com.softwaremill.sttp.Uri = http://[abc]
```

Looking at sources, following logic is happening for host translation:
 ```scala
private val IpV6Pattern = "[0-9a-fA-F:]+".r

  private def encodeHost: String =
    host match {
      case IpV6Pattern() => s"[$host]"
      case _             => UriCompatibility.encodeDNSHost(host)
}
```

Is there any particular reason why sttp cares at all about IPv6?

**Comment by adamw** [comment]
The address has to be properly formatted into an URL. Let's say you have `val host = "::1"`, and you'd like to create an address with that.

So you write:

```
uri"http://$host/something"
```

and expect this to be serialised to a string as `http://[::1]/something`. Hence the regexp.

However, I suspect it might be improved to catch only real ipv6 addresses :). If you'd have a suggestion for improvement, please create a PR!

**Comment by adamw** [comment]
Well, but then if I have a configuration file with the host to use, I would expect things to work the same regardless if I enter an ipv4 or ipv6 address. So I'd keep the functionality, just improve the detection

**Comment by simonnineone** [comment]
Fine, I agree that compat is more important here. :) I'll try submitting some PR soon.

**Comment by adamw** [comment]
Released in 1.5.13
